jujugui/static/gui/src/app/store/env/api.js

```diff
@@ -595,7 +599,7 @@ YUI.add('juju-env-api', function(Y) {
       var cback = function(err, response) {
         this.handleLogin({error: err, response: response});
         if (callback) {
-          callback(err);
+          callback(err, response);
```

Comment:
I am not sure I like that the full response is passed to callers. If the goal is only debugging, then we can log the response here in case of errors, or include a serialized version of the response in the returned error. I know the response is not used by callers, but nothing prevents them to use it, and theoretically we are coupling all the places in which this API is called to the response from the server, which should only be handled in one place, here.